stix2/properties.py

```diff
@@ -106,23 +139,72 @@ def __init__(self, type):
         self.required_prefix = type + "--"
         super(IDProperty, self).__init__()
 
-    def validate(self, value):
-        # TODO: validate GUID as well
+    def clean(self, value):
         if not value.startswith(self.required_prefix):
             raise ValueError("must start with '{0}'.".format(self.required_prefix))
+        try:
+            uuid.UUID(value.split('--', 1)[1], version=4)
+        except Exception:
+            raise ValueError("must have a valid version 4 UUID after the prefix.")
         return value
 
     def default(self):
         return self.required_prefix + str(uuid.uuid4())
 
 
+class IntegerProperty(Property):
+
+    def clean(self, value):
+        try:
+            return int(value)
+        except Exception:
+            raise ValueError("must be an integer.")
+
+
 class BooleanProperty(Property):
-    # TODO:  Consider coercing some values (like the strings "true" and "false")
 
-    def validate(self, value):
-        if not isinstance(value, bool):
-            raise ValueError("must be a boolean value.")
-        return value
+    def clean(self, value):
+        if isinstance(value, bool):
+            return value
+
+        trues = ['true', 't']
+        falses = ['false', 'f']
+        try:
+            if value.lower() in trues:
+                return True
+            if value.lower() in falses:
+                return False
+        except AttributeError:
+            if value == 1:
+                return True
+            if value == 0:
+                return False
+
+        raise ValueError("must be a boolean value.")
+
+
+class TimestampProperty(Property):
+
+    def clean(self, value):
+        if isinstance(value, dt.date):
+            if hasattr(value, 'hour'):
+                return value
+            else:
+                # Add a time component
+                return dt.datetime.combine(value, dt.time(), tzinfo=pytz.timezone('US/Eastern'))
```

Comment:
Also, pytz timezones [shouldn't be passed](http://pytz.sourceforge.net/#localized-times-and-date-arithmetic) to `tzinfo` of datetime constructors: 

> Unfortunately using the tzinfo argument of the standard datetime constructors ‘’does not work’’ with pytz for many timezones.